dotcom-rendering/src/web/components/PinnedPost.importable.tsx

```diff
@@ -108,6 +111,41 @@ export const PinnedPost = ({
 	const [expanded, setExpanded] = useState(false);
 	const [showButton, setShowButton] = useState(false);
 
+	const hasBeenSeen = useRef(false);
+	const timingRef = useRef<{
+		start: () => void;
+		end: () => number;
+		clear: () => void;
+	} | null>(null);
+
+	const [isIntersecting, setRef] = useIsInView({
+		threshold: 0.5,
+		repeat: true,
+	});
+
+	useEffect(() => {
+		if (!timingRef.current) {
+			timingRef.current = initPerf('pinned-post-view-duration');
+		}
+
+		if (isIntersecting) {
+			hasBeenSeen.current = true;
+			timingRef.current.clear();
+			timingRef.current.start();
+		} else if (hasBeenSeen.current) {
+			const duration = timingRef.current.end() / 1000;
+			console.log(`event sent with dutration ${duration}`);
+			submitComponentEvent({
+				component: {
+					componentType: 'CONSENT',
```

Comment:
This component type was temprily used because the actual component type (pinned-post-view-duration) is come in a newer version of guardian/libs that is waiting to be merged on this PR  https://github.com/guardian/dotcom-rendering/pull/4141